src/parser/Parser.ts

```diff
@@ -2847,6 +3006,17 @@ export interface LocalVarEntry {
     type: BscType;
 }
 
+export interface TokenWithIndex {
+    token: Token;
+    index: number;
+    isUnknown?: boolean;
```

Comment:
What does this mean? is unknown token? index is unknown? If index is unknown, how about we just set index to undefined instead.